### DISCUSSION TITLE
StormSim Functionality DLL Conversion

### DISCUSSION THREAD

**Discussion by fgarcia0128** [coastal-hazards-system/StormSim-Library#86] [Announcements]
Opening up this issue to have a place where we can document/discuss DLL conversion of StormSim specific functionality to adress the needs for CHART. First attemp will be converting Eurotop function into a DLL to hand over to Will's team to test out cloud compute framework.

**Starting Point Resources:**
1. Automatically Converting MATLAB Code to C Code (2014)

Attend this presentation to learn how to automatically generate readable and portable C code from your MATLAB® algorithms. With MATLAB Coder™, you can reduce development time by smoothly transitioning from algorithm to implementation using automatic C code generation. This C code does not require any further MATLAB libraries and can be freely distributed.

https://www.mathworks.com/videos/automatically-converting-matlab-code-to-c-code-96483.html

2. MATLAB to C/C++ Made Easy (2020)

In this webinar, we demonstrate the workflow for generating readable and portable C/C++ code from your MATLAB algorithms using MATLAB Coder. Using the command line approach or the MATLAB Coder app, you can introduce implementation requirements to your algorithms written in MATLAB and generate readable source code, standalone compiled executable, or a library that can be shared across your organization.

https://www.mathworks.com/videos/matlab-to-c-made-easy-81870.html

**Comment by fgarcia0128** [comment]
@JacobCreamer Please make use of this thread for discussion points/doubts/leasons learned.

**Comment by KHodgens** [comment]
Thanks much for setting this up Fabian!

Here was the other webpage on *.dll's: https://www.mathworks.com/help/coder/ug/deploy-generated-code.html

**Comment by JacobCreamer** [comment]
It looks like I'll need MATLAB Coder to perform any conversion to C++, and then work from there for DLL conversion. Did I hear correctly that I should contact Madison if I need the Coder app?

**Comment by fgarcia0128** [comment]
Yes

**Comment by fgarcia0128** [comment]
Deprecated functionality
